src/plot_api/subroutines.js

```diff
@@ -124,31 +124,20 @@ exports.lsInner = function(gd) {
 
         var xDomain = plotinfo.xaxis.domain;
         var yDomain = plotinfo.yaxis.domain;
-        var plotgroupBgData = [];
+        var plotgroup = plotinfo.plotgroup;
+        var plotgroupBg;
 
         if(overlappingDomain(xDomain, yDomain, lowerDomains)) {
-            plotgroupBgData = [0];
-        }
-        else {
+            var pgNode = plotgroup.node();
+            plotgroupBg = plotinfo.bg = Lib.ensureSingle(plotgroup, 'rect', 'bg');
+            pgNode.insertBefore(plotgroupBg.node(), pgNode.childNodes[0]);
+        } else {
+            plotgroupBg = plotgroup.select('rect,bg');
```

Comment:
`rect.bg` perhaps?

Reply:
Nice catch. I guess that means we have some untested code branches in there :rage:

Reply:
> we have some untested code branches in there

maybe, maybe not. `rect,bg` would delete all `rect` elements (are there any others? if not, then what you have there will work, it just won't be as efficient) and all `bg` elements (which don't exist)

Reply:
fixed in 5307125ef9655bec0c2e24ba444b3cfc4c1c4f48